### ISSUE TITLE
document scanner rotates the image when it recognizes the document

### ISSUE THREAD

**Issue by Tavorc** [googlesamples/mlkit#798]
When i'm using the ML kit for document scanner, most of the time(like 95%), the document that's recognized by the library it rotates the image.

any idea how to solve it?
Does it happen to someone?
![Screenshot_20240326_095202_Google Play services](https://github.com/googlesamples/mlkit/assets/15210081/c77ad32b-f64f-4440-b452-649345f65451)

**Comment by listvin** [comment]
Does your unwanted rotation happen with LTR scripted docs? Are all of your docs receipts?

I suspect that MLKit's document scanner "UI flow" may be slightly tied with Text Recognition API, which in turn [does not support Hebrew or any other RTL scripts](https://developers.google.com/ml-kit/vision/text-recognition/v2/languages). Even if it's irrelevant..

As you may have noticed in real life people who don't know Hebrew are trying to read documents written in Hebrew upside down. Idk about other rtl scripts irl, but I guess the fact that almost all Hebrew letters have the same height does not help at all.

These are just thots, I am not affiliated with Google in anyway. **I believe it's indeed a bug** since API seems to be designed text-agnostic. Especially in manual mode.

If all of your data are receipts, probably you can postprocess them on low level or with tesseract

**Comment by listvin** [comment]
Inspired by:

https://github.com/googlesamples/mlkit/issues/784#issuecomment-2004640133

It seems that your Android language is English, can you try switching it to Hebrew?

**Comment by Tavorc** [comment]
first of all thank you.
Yes, all of the docs are receipts, it's fintech app.
I tried to change the language to Hebrew, doesn't work.

there is openCV library that i can use to cropping an image, but i didn't want it because the ML kit is more innovative.

**Comment by Tavorc** [comment]
I think you can know the orientation of the device, for example in the camera there is label 1x that represent the zoom, when i rotate the device the "1X" will rotate also, so probably you can using that.

**Comment by tamirla** [comment]
any update on that ? we're also experiencing the same issue with Flutter library that uses MLKit, when trying to scan documents in Hebrew:

https://github.com/jachzen/cunning_document_scanner/issues/74

**Comment by mebjas** [comment]
Thanks for flagging this issue. We have been working on this and improving our rotation classifier.

@Tavorc @tamirla - is this mostly happening with Hebrew documents or you have faced the issue with english documents as well? 

Would it be possible to share some full image examples (pre-crop, taking form stock camera) where this issues is clearly reproducible (try import from gallery option).

**Comment by tamirla** [comment]
@mebjas Thanks for this update, so far we saw it only for Hebrew documents and only in Android, see attached image as an example

![pic1](https://github.com/user-attachments/assets/1f4615ba-783f-4015-b142-77dfda1fd771)
![pic](https://github.com/user-attachments/assets/804660fb-fb5f-4ca7-b813-af3ef8693a59)

**Comment by mebjas** [comment]
Got it, thanks for sharing!

We have this issue in our radar and this issue is pretty high in our stack rank, however we don't have an active solution for this - and team has prioritised other quality improvement efforts for Q4 2024. 

Will share an update to this in Q1 2025.

**Comment by y-maskin** [comment]
Hi @mebjas, Any updates on this? Facing the same issue with hebrew documents being constantly rotated.
Does anyone find a workaround for this until the fix will be published? (Android Native)

**Comment by Tavorc** [comment]
is there any update?

**Comment by y-maskin** [comment]
@mebjas If the detection of when a rotation is needed or not for Hebrew is complex, Please expose an API to disable the auto rotation.
The library is completely unusable currently for Hebrew.

**Comment by Lampact77** [comment]
https://github.com/googlesamples/mlkit/issues/798#issue-2207498093
How. many countries weren't not allowed to the neighbour country near Palestine for some reasons is the safety excuse but the new to imho not to hate the language or culture but seems there is always reasons to avoid this maybe like what was told that here I was told that we are the terrorist hub but the perception was shared among by the neighbouring country like near the island toward the north or the SCs. near south China Sea. the who deal was jeopardise due to the lost island in the surrounding are known as the island of white rock. the power lies in the tih and teh these batu putih and batu Puteh. bravo stupid people that does not realisze what they have lost

**Comment by mebjas** [comment]
This is definitely one of the top quality gaps in Doc Scanner right now - we are working on sizing this issue and retrain the model to have improved orientation detection for many RTL languages. 

Will share more updates as we have it.

**Comment by frenbergFNX** [comment]
Hi all! We're getting reports from our users about strange rotation issues too. (mostly receipts and invoices in swedish so not rtl). Since our app does not support rtl, in last release we turned off the `android:supportsRtl` setting that was originaly set to true. Could this setting affect the scanner logic somehow!?

**Comment by Lampact77** [comment]
is the box for convenience sake when taking square images such as identity
card. so is easier to control and adjust

**Comment by y-maskin** [comment]
@frenbergFNX Exactly. I believe that providing the ability to disable auto rotation can be a quick win for all of us

**Comment by tamirla** [comment]
@mebjas any updates ?

**Comment by Adi-shahaf** [comment]
any update? I still face the issue

**Comment by logicbaze** [comment]
Currently there are a lot of users experiencing this issue in android. Could we disable auto rotation? Any update on this?

**Comment by shahadat-polygon** [comment]
Any update?

**Comment by shajed99-Polygon** [comment]
**Any update?** I am also facing this same problem. And after spending so much time, I found this.

**Comment by naorzadok** [comment]
Suggested fix: add flag to disable auto-rotation
The auto-rotation is clearly designed for phones held parallel to the ground — a valid use case. However, for RTL documents (Hebrew, Arabic) the rotation classifier consistently misfires, likely due to underrepresentation in training data.
Since the root cause is script-specific, a simple opt-out flag would solve it cleanly:
Kotlin
GmsDocumentScannerOptions.Builder()
    .setAutoRotationEnabled(false)
    .build()
Developers targeting RTL languages could opt out; default behavior stays unchanged for everyone else
